docs(langsmith): SmithDB SDK migration guide [WIP] (#4614)
langchain-ai/docs: src/langsmith/smithdb-sdk-migration.mdx

---
title: Migrate to SmithDB-backed SDK methods
description: Migrate your existing LangSmith SDK methods to their SmithDB-backed equivalents for faster agent observability.
noindex: true
---

import SmithdbMigrationRunsQuery from '/snippets/langsmith/smithdb-migration/runs-query.mdx';
import SmithdbMigrationRunsRetrieve from '/snippets/langsmith/smithdb-migration/runs-retrieve.mdx';

## Context

> In May 2026, we released [SmithDB](https://www.langchain.com/blog/introducing-smithdb?utm_source=docs), a new observability database built from the ground up for modern AI agents. Agent traces have grown dramatically in complexity: hundreds of nested spans, multi-modal content, and spans that remain open for hours. General-purpose databases were never designed to handle these patterns at scale. SmithDB addresses this directly, built in Rust on Apache DataFusion and Vortex with purpose-built support for the query patterns that matter most for agent observability: random access, interactive filtering, full-text search, and tree-aware trace reconstruction.

Reviewer: ```suggestion
In May 2026, we released [SmithDB](https://www.langchain.com/blog/introducing-smithdb?utm_source=docs), a new observability database built for modern AI agents. Agent traces have grown dramatically in complexity: hundreds of nested spans, multi-modal content, and spans that remain open for hours. General-purpose databases were never designed to handle these patterns at scale. SmithDB addresses this directly, built in Rust on Apache DataFusion and Vortex with purpose-built support for the query patterns that matter most for agent observability: random access, interactive filtering, full-text search, and tree-aware trace reconstruction.
```

The response from customers has been outstanding. SmithDB delivers leading performance across the key workloads that matter for agent observability and makes core LangSmith experiences up to 12x faster than before.

> Today we are excited to bring SmithDB-powered methods to all SDK users.

Reviewer: ```suggestion
SmithDB-powered methods are now available to all SDK users.
```

## Deployment support

| Deployment | Status |
|---|---|
| US SaaS | Available — methods are fully SmithDB-backed |
| EU and other SaaS regions | Methods are available but not yet backed by SmithDB |

> | Self-hosted | Coming soon — currently not compatible with new methods |

Reviewer: ```suggestion
| US SaaS | Available. Methods are fully SmithDB-backed. |
| EU and other SaaS regions | Methods are available but not yet backed by SmithDB. |
| Self-hosted | Coming soon. Currently not compatible with new methods. |
```
